subprojects/ide/src/main/java/org/gradle/plugins/ide/api/GeneratorTask.java

```diff
@@ -60,7 +60,13 @@
     protected T domainObject;
 
     public GeneratorTask() {
-        getOutputs().upToDateWhen(Specs.satisfyNone());
+        this(false);
+    }
+    
+    protected GeneratorTask(boolean incremental) {
+        if (!incremental) {
+            getOutputs().upToDateWhen(Specs.satisfyNone());
```

Comment:
Pretty cool, I like it!

Reply:
Could we avoid changing the constructor and push this `getOutputs().upToDateWhen(Specs.satisfyNone());` into the plugins that are not incremental yet?  I think eventually we could make all of them are incremental and make this flag useless.

Also, booleans are evil :)

Reply:
I've changed it so that the constructor signature is the same, but the generator task will tell you whether it's incremental or not.